liquidator/OneInchExchange.js

```diff
@@ -59,38 +90,40 @@ class OneInchExchange {
       });
     }
 
-    // 1 Split config
-    const flags = 0; // Enables all exchanges
-
-    // Number of pieces source volume could be splitted
-    // (Works like granularity, higly affects gas usage.
-    // Should be called offchain, but could be called onchain
-    // if user swaps not his own funds, but this is still considered
-    // as not safe)
-    // More info: https://github.com/CryptoManiacsZone/1inchProtocol#getexpectedreturn
-    const parts = 2;
-
     const expectedReturn = await this.oneSplitContract.methods
-      .getExpectedReturn(fromToken, toToken, amountWei, parts, flags)
+      .getExpectedReturn(fromToken, toToken, amountWei, this.oneInchParts, this.oneInchFlags)
       .call();
 
     const { returnAmount, distribution } = expectedReturn;
 
     this.logger.debug({
       at: "OneInchExchange",
       message: "GetExpectedReturn",
+      currentTime,
       returnAmount,
       distribution
     });
 
+    if (minReturnAmountWei && toBN(returnAmount.toString()).lt(toBN(minReturnAmountWei.toString()))) {
+      this.logger.debug({
+        at: "OneInchExchange",
+        message: "ReturnAmountTooLow",
+        currentTime,
+        returnAmount,
+        minReturnAmountWei
+      });
+      throw new Error("OneInch return amount too low");
+    }
+
     // TODO: Remove hardcoded gas
```

Comment:
You could set `gasLimit` as a class state variable, similar to how the liquidator sets its `txnGasLimit`: https://github.com/UMAprotocol/protocol/blob/master/liquidator/liquidator.js#L61

Reply:
Whats the rational for using gasLimit as a class state variable?

Reply:
We should avoid hardcoding gas limits, and if we're going to do it then we should at least allow the caller to override it via custom configuration

Reply:
As the 1split contract interacts with multiple (and growing) on-chain protocols, I do not recommend using the gasEstimator here as:

- Estimating the gasLimit takes a LOT of time.
- Tha gasEstimator also does a poor job at estimating the actual gas required. [See failed transactions on 1split.eth for ref](https://etherscan.io/address/1split.eth)

Hardcoding gasLimits for specific tasks is [also](https://github.com/makerdao/mcd-cdp-portal/blob/f16bf9d7ad043e910ea66329d0fd963773baa7f2/src/utils/ethereum.js#L136) quite [common](https://github.com/curvefi/curve-dao/blob/curvedao/src/components/deposit/Deposit.vue#L296)

Is a custom config of gasLimits really necessary for a simple task of swapping tokens?

Reply:
> Is a custom config of gasLimits really necessary for a simple task of swapping tokens?
Swaps being simple is inconsistent with the premise that "1split contract interacts with multiple (and growing) on-chain protocols...Estimating the gasLimit takes a LOT of time." Clearly the swap transaction is a very complex process. Why not just estimate the gas? The Liquidator bot will rarely need to swap reserve tokens for synthetic tokens and when it does so it should do so correctly. You also should take advantage of existing `Promise.all`'s and figure out where best to slot in the one inch swapping to make it most efficient.

> Hardcoding gasLimits for specific tasks is also quite common
A lot of users have complained about the super high gas estimation that Curve DAO employs. 

> Tha gasEstimator also does a poor job at estimating the actual gas required. See failed transactions on 1split.eth for ref
This is why we use the `GAS_LIMIT_BUFFER`.